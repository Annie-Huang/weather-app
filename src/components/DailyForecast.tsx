import React from 'react';

/*  <div>
    {day.precipitation} |
    {day.humidity} |
    {day.wind} |
    {day.temperature} |
    {day.weather_description} |
    {day.min_temperature}&#176; |
    {day.max_temperature}&#176; |
    <img src={day.weather_icon} alt=""/> |
    {day.clouds} |
  </div>*/

interface IDailyForecastProp {
  weather_description: string;
  day_of_week: string;
  min_temperature: number;
  max_temperature: number;
  weather_icon: string;
  clouds: string;
}

const DailyForecast: React.FC<IDailyForecastProp> = (props) => (
  <div className='card text-center'>
    <h5 style={{marginBottom: 0}}>{props.day_of_week}</h5>
    <img src={props.weather_icon} alt={props.weather_description} />
    <div>{props.min_temperature}&#176;<span className='text-secondary'>{props.max_temperature}&#176;</span></div>
    <div className='text-secondary'>Clouds: {props.clouds}</div>
  </div>
);


export default DailyForecast;
