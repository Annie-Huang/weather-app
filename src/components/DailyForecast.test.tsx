import DailyForecast, {IDailyForecastProp} from './DailyForecast';
import React from 'react';
// @ts-ignore
import { shallow } from 'enzyme';

test('CurrentWeather', () => {
  const props: IDailyForecastProp = {
    weather_description: "weather_descriptionStr",
    day_of_week: "day_of_weekStr",
    min_temperature: 10,
    max_temperature: 20,
    weather_icon: "weather_iconStr",
    clouds: "cloudsStr",
  }

  const wrapper  = shallow(<DailyForecast {...props} />)
  expect(wrapper).toMatchSnapshot();
});
