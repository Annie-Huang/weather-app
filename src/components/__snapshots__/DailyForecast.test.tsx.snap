// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentWeather 1`] = `
<div
  className="card text-center"
>
  <h5
    style={
      Object {
        "marginBottom": 0,
      }
    }
  >
    day_of_weekStr
  </h5>
  <img
    alt="weather_descriptionStr"
    src="weather_iconStr"
  />
  <div>
    10
    °
    <span
      className="text-secondary"
    >
      20
      °
    </span>
  </div>
  <div
    className="text-secondary"
  >
    Clouds: 
    cloudsStr
  </div>
</div>
`;
